### PR TITLE
[DONE] SnackBar integration

### PR DESCRIPTION
**Pull request by gijs**
Screenshot (see below for details):

jun-21-2017 14-32-40

The SnackBar component is meant to be the notification system of this app.

This PR adds the following:

- Fixes the styling and animation of the SnackBar component
- Connects the SnackBar to Redux
- For demo purposes, shows SnackBar on componentDidMount of Map/List views
- Updates the Storybooks for the SnackBar component
- Adds this to the [state plan document](https://github.com/nens/G4AW-lizard-client/blob/8b4166b82e257ec1ea251f21d23140b2d2939403/G4AW-state-plan.md)

The SnackBar component is activated by calling the `showSnackBar` dispatcher:

```js
    this.props.showSnackBar({
      message: "Map view",
      subMessage: "This is the map view",
      autoHideDuration: 4000
    });
```

The `autoHideDuration` prop is optional and is in miliseconds. 

If omitted, the user will have to click OK to dismiss the notification. If provided, the `HIDE_SNACKBAR` action is dispatched after the provided number of ms.

### CODE REVIEW

File: src/components/SnackBar.jsx [hunk 1]
```diff
@@ -12,56 +12,37 @@ import { VelocityComponent } from "velocity-react";
...
-          opacity: this.state.showComponent ? 1 : 0
-        }}
-      >
-        <SnackBarContent {...this.props} />
-      </VelocityComponent>
+      <div className={styles.SnackBarWrapper}>
+        <VelocityComponent
+          duration={150}
+          animation={{
+            translateY: isOpen ? 0 : "100px"
```
Comment (gijs) [inline src/components/SnackBar.jsx]: note to self: `"100px"` in JSX may just be written as `100` but they both work fine

File: src/reducers/Ui.js [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { initialUiState } from "../store/Store";
 
 export default function(state = initialUiState, action) {
   switch (action.type) {
+    case ActionTypes.SHOW_SNACKBAR:
+      return { ...state, showSnackBar: true, snackBarOptions: action.options };
+    case ActionTypes.HIDE_SNACKBAR:
+      return { ...state, showSnackBar: false, snackBarOptions: {} };
```
Comment (d-weeteling) [inline src/reducers/Ui.js]: Hier wordt snackBarOptions naar ``{}` gezet terwijl die initieel op `{message: "", subMessage: ""}` gezet wordt: dat is inconsistent.

File: src/store/Store.js [hunk 3]
```diff
@@ -15,7 +15,12 @@ createStoreWithMiddleware = applyMiddleware(thunkMiddleware, logger)(
 // See G4AW-state-plan.md.
 
 export const initialUiState = {
-  currentView: "ListSearchView"
+  currentView: "ListSearchView",
+  showSnackBar: false,
+  snackBarOptions: {
+    message: "",
+    subMessage: ""
+  }
```
Comment (d-weeteling) [inline src/store/Store.js]: Waarom op lege string initialiseren als je ze ook op `null` kunt zetten?

File: stories/SnackBarStories.jsx [hunk 4]
```diff
@@ -8,24 +8,15 @@ export default function runSnackBar() {
   storiesOf("SnackBar", module)
     .add("default", () => (
       <SnackBar
-        open={true}
+        isOpen={true}
```
Comment (d-weeteling) [inline stories/SnackBarStories.jsx]: Beter schrijf je `isOpen={true}` gewoon als `isOpen`, zo leerde jij mij laatst ;)

File: src/actions/UiActions.js [hunk 5]
```diff
@@ -1,6 +1,29 @@
...
+let timerAutoHideId;
 
 export function changeView(dispatch, newView) {
-  const changeViewAction = { type: CHANGE_VIEW, newView };
-  dispatch(changeViewAction);
+  dispatch({ type: CHANGE_VIEW, newView });
+}
+
+export function showSnackBar(dispatch, options) {
+  const { showSnackBar } = theStore.getState().ui;
```
Comment (gijs) [inline src/actions/UiActions.js]: not needed

File: src/actions/UiActions.js [hunk 6]
```diff
@@ -1,6 +1,29 @@
...
+  if (options.autoHideDuration) {
+    clearTimeout(timerAutoHideId);
+    timerAutoHideId = setTimeout(() => {
+      dispatch(hideSnackBar);
+    }, options.autoHideDuration);
+  } else {
```
Comment (gijs) [inline src/actions/UiActions.js]: else not needed

File: src/components/SnackBar.jsx [hunk 7]
```diff
@@ -12,56 +12,37 @@ import { VelocityComponent } from "velocity-react";
 class SnackBar extends Component {
   constructor() {
     super();
-    this.state = {
-      showComponent: true
-    };
   }
   componentDidMount() {
-    if (this.state.showComponent && this.props.autoHideDuration) {
+    if (this.props.autoHideDuration) {
```
Comment (gijs) [inline src/components/SnackBar.jsx]: mag helemaal weg

File: src/store/Store.js (hunk 3 above)
Comment (gijs) [inline src/store/Store.js]: te overwegen

**Pull request merged**